src/executable-code/executable-fragment.js

```diff
@@ -181,6 +188,57 @@ export default class ExecutableFragment extends ExecutableCodeTemplate {
     }
   }
 
+  removeAllOutputNodes() {
+    const outputNode = this.element.getElementsByClassName("code-output").item(0)
+    while (outputNode && outputNode.lastChild) {
+      outputNode.removeChild(outputNode.lastChild)
+    }
+  }
+
+  applyStateUpdate(stateUpdate) {
+    if (stateUpdate.errors === null) {
+      stateUpdate.errors = []
+    } else if (stateUpdate.errors !== undefined) {
```

Comment:
Well, I placed this peace of code after filtration but could not come up with any idea how to make it simpler. We must distinguish null here because passing null as `stateUpdate.errors` tells the function to remove previous errors in the state. Therefore `if` is necessary here.